src/test/java/web/tests/FlipkartTest.java

```diff
@@ -0,0 +1,103 @@
+package web.tests;
+
+import com.codeborne.selenide.Configuration;
+import com.codeborne.selenide.ElementsCollection;
+import com.codeborne.selenide.Selenide;
+import com.codeborne.selenide.SelenideElement;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import web.pages.flipkart.HomePage;
+import web.pages.flipkart.SearchPage;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Iterator;
+import static com.codeborne.selenide.Selenide.$$x;
+import static com.codeborne.selenide.Selenide.$x;
+
+public class FlipkartTest {
+    private Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    @BeforeEach
+    public void setUp(){
+        Configuration.baseUrl="https://www.flipkart.com";
+        Selenide.open("");
+    }
+
+    @Test
+    public void getDetails()  {
+        String product ="shoes";
+        String filter="Price -- Low to High";
+        System.out.println(product);
+        System.out.println(filter);
+        HomePage homepage=new HomePage();
+        SearchPage searchpage=new SearchPage();
+        product=product.trim();
+        logger.info(product);
+        //to handle popup
+        homepage.popUpCancel();
+        //to enter product i.e shoes in textfield
+        homepage.setShoes(product);
+        //to enter search button
+        homepage.searchShoes();
+        delay();
+        //to select low to high filter option
+        sortShoes(filter);
+        delay();
+        //to do the pagination of page1
+        counting();//counting prices of page1
+        delay();
+        //pagination for page2
+        searchpage.selectSecondPage();
+        delay();
+        counting();//counting prices of page2
+
+    }
+    public void sortShoes(String filter) {
```

Comment:
Please add some description for this function.
Also wouldn't it make more sense to add this to your Page class?